### DISCUSSION TITLE
452 4.5.3 Error: too many recipients

### DISCUSSION THREAD

**Discussion by ichiy** [PHPMailer/PHPMailer#2325] [General]
I created a code based on PHPMailer to send bulk emails for marketing purposes and I used `for` statement to loop through the list of recipients . The code works fine and PHPMailer is awesome, However whenever I try to send to more then 25 recipients I get the following error **452 4.5.3 Error: too many recipients** to solve this I tried to split the distribution list in two or more arrays and I tried also to delay execution of the script using different methods and it did not work for me. I hope anyone can help me in the community to create  a solution in code.

Ps: I cannot contact ISP due to the permissions of my post.

**Comment by Synchro** [comment]
It sounds like you're already doing all that you can. It might be time to find a better hosting provider or invest in dedicated SMTP capacity. If you can't contact your hosting provider for support, you *definitely* need a new one; anybody responsible for sending large volumes of email should have the ability to communicate with system admins. This is really outside PHPMailer's area of responsibility though.

You could install a local mail server and get it to relay for you, which will remove all of your immediate sending restrictions, and the mail server will then deal with ongoing limits and retries/deferrals for you. Mail servers are built to do this stuff, and they are better at it than any PHP code that you write.
